src/pytorch_lightning/trainer/trainer.py

```diff
@@ -527,17 +532,19 @@ def _setup_on_init(self) -> None:
         self.should_stop = False
         self.state = TrainerState()
         self.num_training_batches = float("inf")
-        self.train_dataloader = None
-
-        self.num_sanity_val_batches = []
-        self.num_test_batches = []
-        self.num_val_batches = []
-        self.num_predict_batches = []
-        self.test_dataloaders = None
-        self.val_dataloaders = None
-        self.predict_dataloaders = None
-        self._last_train_dl_reload_epoch = None
-        self._last_val_dl_reload_epoch: Optional[int] = None
+
+        self.train_dataloader: Optional[Union[CombinedLoader, TRAIN_DATALOADERS]] = None
+
+        self.num_sanity_val_batches: List[Union[int, float]] = []
+        self.num_test_batches: List[Union[int, float]] = []
+        self.num_val_batches: List[Union[int, float]] = []
+        self.num_predict_batches: List[Union[int, float]] = []
+
+        self.test_dataloaders: Optional[List[DataLoader]] = None
+        self.val_dataloaders: Optional[List[DataLoader]] = None
+        self.predict_dataloaders: Optional[List[DataLoader]] = None
+        self._last_train_dl_reload_epoch = float("-inf")
+        self._last_val_dl_reload_epoch = float("-inf")
```

Comment:
In data_connector there is a check `self.trainer._last_train_dl_reload_epoch is None`, wouldn't that one always be false now? can be deleted probably?

similarly, also for the _last_val_dl_reload_epoch

Reply:
Good catch!